duckduckgo/duckduckgo-privacy-extension
# New privacy grade mvp

**Reviewer:** @MariagraziaAlastra 

**CC:** @russellholt 

## Description:

Plugs in the new privacy grade class.

## Steps to test this PR:

Run `npm install`, then:

1. Test specific sites. Pick several sites. Here's how to test the grades are being set correctly:

- Open the console on the background page. Set a console filter for "Grade".
- Open the site in question
- Verify grade
    - Check the Grade object in the background page console log - open "scores" and note what grades are listed there
    - Check that the grade is the same across the toolbar icon, the rating hero, and the grade details subview (at the bottom)
- Verify trackers
    - Check the "entitiesBlocked" and the "entitiesNotBlocked" entries in the background tab console.log
    - Each of the entries in "entitiesBlocked" should be listed as a network in the Trackers subview
    - If there's more than one entry in "entitiesNotBlocked", it would be in the Trackers subview with a (not blocked) on it. (If there's one entity there, it might refer to the site's own parent entity, so it might not be listed in the Trackers subview)
    - In the grade details subview the number of major tracker networks should include all "entitiesBlocked" that have a prevalence of over 7% (the number may be higher than current prod)
    - If the site's own parent tracker prevalence is over 7%, the grade details subview should say "site is a major tracker network"
- Verify HTTPS
    - If the site supports HTTPS, the https portion of the site and enhanced scores should say 0
    - Otherwise it should say 10
- Verify privacy practices
    - If we have TOSDR info on the site, it should be shown in the privacy practices subview as it currently is in production
    - If we list the privacy practices for the site as "bad", the privacy grade should have a privacy score of 7 or 10
    - If we list the privacy practices for the site as "mixed" (e.g. for Github) the privacy score should be 1
    - If we list the privacy practices as "good" the privacy score should be 0
    - If we list the privacy practices as "unknown" the privacy score should be 2

2. Test that special rating states ("calculating", special pages, etc) still work correctly
3. Test that element hiding still works on Oath sites (just checking it still works should be good)

## Automated tests:

###### Reviewer Checklist:


###### PR Author Checklist:
